export_rsvps.py

```diff
@@ -3,69 +3,58 @@
-    'DB_USER': 'Username for database connection.',
-    'EXTRA_WHERE': 'Anything to add to the WHERE clause in the RSVP query.',
-    'KEY': 'Key to validate.',
-    'MAX_AGE': 'Number of days a key should be considered valid.',
-    'SECRET': 'Secret to use for validation.',
+    'KEY': 'Key to validate.'
 }
 
-REQUIRED_ARGS = [
-    'DB_HOST', 'DB_NAME', 'DB_PASS', 'DB_PORT', 'DB_USER', 'KEY', 'MAX_AGE',
-    'SECRET', 
-]
+REQUIRED_ARGS = ['KEY']
 
 
 def main(args):
-    key = validate_key.main(args)
+    script_settings = get_secret('ak-partner-rsvp')
+    db_settings = get_secret('redshift-admin')
+    db_schema = script_settings['DB_SCHEMA']
+    extra_where = script_settings['EXTRA_WHERE'] or ''
+
+    key = validate_key.main(args, script_settings)
+
     if key.get('valid', False):
         connection = psycopg2.connect(
-            host=args.DB_HOST,
-            port=args.DB_PORT,
-            user=args.DB_USER,
-            password=args.DB_PASS,
-            database=args.DB_NAME
+            host=db_settings['host'],
+            port=db_settings['port'],
+            user=db_settings['username'],
+            password=db_settings['password'],
+            database=db_settings['dbName']
         )
         cursor = connection.cursor(
             cursor_factory=psycopg2.extras.RealDictCursor
         )
-        query = """
-        SELECT
-            u.email, u.first_name, u.middle_name, u.last_name, u.state, u.city,
-            u.zip, MIN(a.created_at) AS action_datetime, MAX(s.role) AS role
-        FROM %s.core_user u
-        JOIN %s.events_eventsignup s ON s.user_id = u.id
-        JOIN %s.events_event e ON e.id = s.event_id
-        JOIN %s.events_campaign c ON c.id = e.campaign_id
-        LEFT JOIN %s.core_action a ON (
-            a.page_id = s.page_id
-            AND a.user_id = u.id
-        )
-        WHERE c.name = %s
-        %s
-        AND a.source = %s
-        GROUP BY 1,2,3,4,5,6,7""" % (
-                    args.DB_SCHEMA,
-                    args.DB_SCHEMA,
-                    args.DB_SCHEMA,
-                    args.DB_SCHEMA,
-                    args.DB_SCHEMA,
-                    '%s',
-                    args.EXTRA_WHERE,
-                    '%s'
-        )
-        cursor.execute(query, (key.get('campaign', ''), key.get('source', '')))
+
+        query = f"""
+            SELECT
+                u.email, u.first_name, u.middle_name, u.last_name, u.state, u.city,
+                u.zip, MIN(a.created_at) AS action_datetime, MAX(s.role) AS role
+            FROM {db_schema}.core_user u
+            JOIN {db_schema}.events_eventsignup s ON s.user_id = u.id
+            JOIN {db_schema}.events_event e ON e.id = s.event_id
+            JOIN {db_schema}.events_campaign c ON c.id = e.campaign_id
+            LEFT JOIN {db_schema}.core_action a ON (
+                a.page_id = s.page_id
+                AND a.user_id = u.id
+            )
+            WHERE c.name = '{key.get('campaign', '')}'
```

Comment:
This was intentionally not included before - by passing it in as a parameter it's escaped, preventing issues with SQL injection.

Reply:
woops, will revert